server/service/src/requisition_line/response_requisition_line/delete.rs

```diff
@@ -0,0 +1,184 @@
+use repository::{
+    requisition_row::RequisitionType, RepositoryError, RequisitionLineRowRepository,
+    StorageConnection,
+};
+
+use crate::{
+    requisition::common::{check_approval_status, check_requisition_row_exists},
+    requisition_line::common::check_requisition_line_exists,
+    service_provider::ServiceContext,
+};
+
+#[derive(Debug, PartialEq, Clone, Default)]
+pub struct DeleteResponseRequisitionLine {
+    pub id: String,
+}
+
+#[derive(Debug, PartialEq)]
+
+pub enum DeleteResponseRequisitionLineError {
+    RequisitionLineDoesNotExist,
+    NotThisStoreRequisition,
+    NotAResponseRequisition,
+    RequisitionDoesNotExist,
+    CannotEditRequisition,
+    DatabaseError(RepositoryError),
+}
+
+type OutError = DeleteResponseRequisitionLineError;
+
+pub fn delete_response_requisition_line(
+    ctx: &ServiceContext,
+    input: DeleteResponseRequisitionLine,
+) -> Result<String, OutError> {
+    ctx.connection
+        .transaction_sync(|connection| {
+            validate(connection, &ctx.store_id, &input)?;
+
+            RequisitionLineRowRepository::new(connection)
+                .delete(&input.id)
+                .map_err(OutError::DatabaseError)
+        })
+        .map_err(|error| error.to_inner_error())?;
+    Ok(input.id)
+}
+
+fn validate(
+    connection: &StorageConnection,
+    store_id: &str,
+    input: &DeleteResponseRequisitionLine,
+) -> Result<(), OutError> {
+    let requisition_line_row = check_requisition_line_exists(connection, &input.id)?
+        .ok_or(OutError::RequisitionLineDoesNotExist)?
+        .requisition_line_row;
+
+    let requisition_row =
+        check_requisition_row_exists(connection, &requisition_line_row.requisition_id)?
+            .ok_or(OutError::RequisitionDoesNotExist)?;
+
+    if requisition_row.store_id != store_id {
+        return Err(OutError::NotThisStoreRequisition);
+    }
+
+    if requisition_row.r#type != RequisitionType::Response {
+        return Err(OutError::NotAResponseRequisition);
+    }
+
+    if check_approval_status(&requisition_row) {
+        return Err(OutError::CannotEditRequisition);
+    }
```

Comment:
This error is supposed to check for status since you cannot delete a requisition that was created from an Internal Order so approval status doesn't really matter.

```suggestion
    if requisition_row.status == RequisitionStatus::Finalised
        return Err(OutError::CannotEditRequisition);
    }
```